enhancements/machine-api/azure-ultra-disks.md

```diff
@@ -72,59 +72,105 @@ The upstream design is followed as closely as possible to ensure smoother future
 
 A new field will be added to the `AzureMachineProviderSpec` struct `DataDisks`. The interface will be a slice of `DataDisk` structs allowing multiple Data Disks and various fields specific to them to be further specified.
 
-A `DataDisk` has one Composite type field, `ManagedDisk` of type `ManagedDiskParameters`, which is a type already defined in the Machine API spec, used for the `ManagedDisk` field in the `OSDisk` type struct, another `AzureMachineProviderSpec` field.
+`DataDisk` will need a slightly modified version of the `ManagedDiskParameters` type (due to an extra `UltraSSD_LRS` possible value) for its `ManagedDisk` field.
 
+For this reason a new Composite type field `DataDiskManagedDiskParameters` will be added and the original `ManagedDiskParameters`, used in `OSDisk`, will be renamed to `OSDiskManagedDiskParameters`.
+
+A new `StorageAccountType` type that describes the possible values for the `StorageAccountType` field on `DataDisk` will be added.
+
+A new `CachingTypeOption` type that describes the possible values for the `CachingType` field on `DataDisk` will also be added.
 
 ```go
 type AzureMachineProviderSpec struct {
   // Existing fields will not be modified
   ...
 
-  // DataDisk specifies the parameters that are used to add one or more data disks to the machine
+  // DataDisk specifies the parameters that are used to add one or more data disks to the machine.
   // +optional
   DataDisks []DataDisk `json:"dataDisks,omitempty"`
 }
 
 // DataDisk specifies the parameters that are used to add one or more data disks to the machine.
+// A Data Disk is a managed disk that's attached to a virtual machine to store application data.
+// It differs from an OS Disk as it doesn't come with a pre-installed OS, and it cannot contain the boot volume.
+// It is registered as SCSI drive and labeled with the chosen `lun`. e.g. for `lun: 0` the raw disk device will be available at `/dev/disk/azure/scsi1/lun0`.
+//
+// As the Data Disk disk device is attached raw to the virtual machine, it will need to be partitioned, formatted with a filesystem and mounted, in order for it to be usable.
+// This can be done by creating a custom userdata Secret with custom Ignition configuration to achieve the desired initialization.
+// At this stage the previously defined `lun` is to be used as the "device" key for referencing the raw disk device to be initialized.
+// Once the custom userdata Secret has been created, it can be referenced in the Machine's `.providerSpec.userDataSecret`.
+// For further guidance and examples, please refer to the official OpenShift docs.
```

Comment:
just curious if there is a link we can give here?

Reply:
There is a link but it's not specific for Data Disks or Azure, rather for AWS: https://docs.openshift.com/container-platform/4.10/post_installation_configuration/node-tasks.html#machine-node-custom-partition_post-install-node-tasks.
My intention is to have a chat with @jeana-redhat on how we can best proceed on the docs side and what to link from where :) If we decide on creating a specific doc section we could link it here and in the api godoc.

Reply:
Cool - guess we will see if it is different enough to need its own heading or is AWS and Azure can use the same process with different details

Reply:
Sounds like a plan :)

Reply:
+1, sounds great